components/UI/RangeHistogram/components/BarChart.tsx

```diff
@@ -1,63 +1,57 @@
-import React, { useEffect, useRef } from 'react';
+import React, { useEffect, useState } from 'react';
 import classnames from 'classnames';
 import { getPercent } from '../utils';
 import { HistogramData, MinMax, Range } from '../types';
 import barchartStyles from './BarChart.module.css';
 
 interface Props {
-    data: HistogramData;
-    range: MinMax;
     height: number;
+    data: HistogramData;
+    minValue: number;
+    maxValue: number;
     onSelect?: (item: Range) => void;
+    sliderMinMax: MinMax;
 }
 
-export function BarChart({ data, range, height, onSelect }: Props) {
+export function BarChart({ data, sliderMinMax, height, onSelect, minValue, maxValue }: Props) {
+    const [innerSelectedMinMax, setInnerSelectedMinMax] = useState<MinMax>();
+
+    const [actualMinMax, setActualMinMax] = useState<MinMax>({
+        min: minValue,
+        max: maxValue,
+    });
+
+    useEffect(() => {
+        setActualMinMax(innerSelectedMinMax);
+    }, [innerSelectedMinMax]);
+
+    useEffect(() => {
+        setActualMinMax(sliderMinMax);
+    }, [sliderMinMax]);
+
     const total = data.reduce((acc, item) => acc + item.value, 0);
     const max = Math.max(...data.map((item) => item.value));
 
     const items = data.map((item) => ({
         ...item,
         percent: getPercent(0, total, item.value),
         height: getPercent(0, max, item.value),
-        isActive: item.from >= range.min && item.to <= range.max,
+        isActive: item.from >= actualMinMax.min && item.to <= actualMinMax.max,
     }));
 
-    const selected = useRef(false);
-
-    const onMouseDown = (item) => {
```

Comment:
maybe we can return this behaviour?
it helps to select items from mobile, you can swipe full histogram area, not only range

Reply:
returned! you can check it